src/exceptions/ErrorHandler.js

```diff
@@ -25,7 +25,9 @@ class ErrorHandler{
 	wrapBarcodeCall(func){
 		try{
 			var result = func(...arguments);
-			this.api._options.valid(true);
+			if (typeof this.api._options.valid !== 'undefined') {
```

Comment:
Oh, you can do it in old specs as I see (http://www.ecma-international.org/ecma-262/5.1/#sec-15.1.1.3). But for a public library your way is better :)